tests/epyccel/test_epyccel_variable_annotations.py

```diff
@@ -166,3 +174,82 @@ def homogeneous_tuple_annotation():
 
     assert epyc_homogeneous_tuple_annotation() == homogeneous_tuple_annotation()
     assert isinstance(epyc_homogeneous_tuple_annotation(), type(homogeneous_tuple_annotation()))
+
+@pytest.mark.parametrize( 'language', [
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("fortran", marks = [
+            pytest.mark.skip(reason="Variable declaration not implemented in fortran"),
+            pytest.mark.fortran]),
+        pytest.param("python", marks = pytest.mark.python)
+    ]
+)
+def test_homogeneous_set_annotation_int(language):
+    def homogeneous_set_annotation ():
+        a : set[int] #pylint: disable=unused-variable
```

Comment:
This is only valid in Python > 3.8.
3.8 only reaches end of life in November so we still support it. You can use:
```suggestion
        a : 'set[int]' #pylint: disable=unused-variable
```
to get round this